Auto-configure models required by `app.enableAuth`

bajtos opened the pull request:
Modify `app.enableAuth` to automaticaly setup all required models that are not attached to the app nor a datasource.

Users wishing to use this option must provide the name of the data-source to use for these models.

Example usage:

    var app = loopback();
    app.dataSource('db', { connector: 'memory' });
    app.enableAuth({ dataSource: 'db' });

    app.use(loopback.rest());
    app.listen(3000);

The purpose of this new API is to make it easier to setup authentication from code that does not use "slc loopback" project scaffolding, typically from unit-tests.

/to @ritch @raymondfeng please review

raymondfeng commented on lib/application.js:
```diff
@@ -291,10 +291,27 @@ app.dataSources = app.datasources = {};
 
-app.enableAuth = function() {
+app.enableAuth = function(options) {
+  var AUTH_MODELS = ['User', 'AccessToken', 'ACL', 'Role', 'RoleMapping'];
```
What's going to happen if these models are sub-classed?

bajtos replied:
That's a good question. This shortcut is intended only for the simple situations where you typically don't create subclasses. To keep the implementation simple, I can modify the condition determining whether a base model should be attached or not, and do not attach a base model if there is a subclassed model already attached?

From user perspective, if you create let's say Customer extending User:

 - If Customer is attached to a data source or the app, then the builtin User model is left unchanged.
 - If Customer is not attached to the app nor the datasource, then we attach the builtin User model for you.

Thoughts?

raymondfeng replied:
Sounds good.

bajtos commented:
@ritch @raymondfeng ping

ritch commented:
:+1:

bajtos commented:
Reworked.

@raymondfeng PTAL again.

@ritch FYI, I have added a new `loopback` option `loadBuiltinModels` that can be used together with `localRegistry`, see 0ccc1e2.

ritch commented:
> I have added a new loopback option loadBuiltinModels

Cool. We should start marking code like this somehow... so we can easily review it.

bajtos commented:
@slnode test please

ritch commented:
LGTM

raymondfeng commented:
LGTM
